### ISSUE TITLE
Build binaryen.js with CMake

### ISSUE THREAD

**Issue by kripken** [WebAssembly/binaryen#2453]
The current build-js.sh has some issues like compiling all source files sequentially, without parallelization.

One option was https://github.com/WebAssembly/binaryen/pull/2448 which used a Makefile, but that's not windows-friendly.

We should probably use CMake as described there. In fact, we can probably just use the existing CMake we use for native builds here, just with some ifs for the emscripten target. Specifically, for emscripten we'd port only wasm-opt, and we'd include the binaryen.js JS glue etc. (wasm-opt has everything binaryen.js needs, and in fact we may want to use the wasm-opt commandline interface eventually.)

**Comment by kripken** [comment]
I spent a little time trying to do this, but ran into issues I couldn't figure out:

 * I tried to make a subdir, then do `emconfigure cmake ..`. But it still emits build files etc. in the toplevel dir. This seems to make it impossible to have parallel JS and native builds?
 * Even in the toplevel dir, doing that and `emmake make` still calls the native C++ compiler, not emscripten. I guess I need to completely remove the native cached CMake build environment I already had..?
